source/kraken/fill_disruption_from_database.h

```diff
@@ -93,25 +94,28 @@ namespace navitia {
                     impact->id() != const_it["impact_id"].template as<std::string>())) {
                 fill_impact(const_it);
                 last_period_id = "";
-                last_ptobject_id = "";
-                last_message_id = "";
+                pt_object_ids.clear();
                 last_channel_type_id = "";
                 message = nullptr;
                 channel = nullptr;
+                message_ids.clear();
             }
 
             if (impact && (last_period_id != const_it["application_id"].template as<std::string>())) {
                 fill_application_period(const_it);
             }
-            if (impact && (last_ptobject_id != const_it["ptobject_id"].template as<std::string>())) {
+            if (impact
+                    && (pt_object_ids.find(const_it["ptobject_id"].template as<std::string>()) == pt_object_ids.end())) {
```

Comment:
you can use `! pt_object_ids.count(const_it["ptobject_id"].template as<std::string>())`